src/cpp/rtps/resources/ResourceEvent.cpp

```diff
@@ -44,6 +44,10 @@ ResourceEvent::~ResourceEvent()
 
     logInfo(RTPS_PARTICIPANT, "Removing event thread");
     stop_.store(true);
+    {
+        std::lock_guard<TimedMutex> lock(mutex_);
+        cv_.notify_one();
+    }
```

Comment:
From cppreference.com:
```
The notifying thread does not need to hold the lock on the same mutex as the one held by the waiting thread(s); in fact doing so is a pessimization, since the notified thread would immediately block again, waiting for the notifying thread to release the lock. 
```
Suggestion: only notify:
```
stop_.store(true);
cv_.notify_one();
```

Reply:
changed